### ISSUE TITLE
Consider using JsonSerializer directly instead of injecting it with Autofac.

### ISSUE THREAD

**Issue by dawidkomorowski** [dawidkomorowski/geisha#214]
Serialization seems to be very deep implementation detail of certain features. Classes using serialization typically abstracts away details of data storage or data transport, internally using serialization for that purpose. There is doubtful benefit of injecting serializer to those classes and testing them in isolation - without actual serializer.

Most likely code could be simplified if classes using serialization would keep it as implementation detail and be tested with integration tests doing actual serialization.

Usefulness of common serialization helper class with predefined configuration is also in question. Evaluate value of maintaining helper class instead of direct usage of serializer. Helper class abstracts library used - is it worth it?

Also consider migrating from `Newtonsoft.Json` to `System.Text.Json` for standardization and performance reasons. https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to

~~It is not clear if it should be implemented before or after #192.~~
This issue should be implemented after #192 so support for polymorphic serialization/deserialization is no longer needed.

Acceptance criteria:
- [x] Usages of `IJsonSerializer` and `JsonSerializer` from `Geisha.Common.Serialization` are replaced with `System.Text.Json.JsonSerializer`.
- [x] `IJsonSerializer` and `JsonSerializer` in `Geisha.Common.Serialization` are removed.
- [x] Dependency on `Newtonsoft.Json` is removed.
